### PR TITLE
feat(ts): document renderingContent

### PR DESCRIPTION
**Pull request by Haroenv**
DX-2066

see also the helper PR: https://github.com/algolia/algoliasearch-helper-js/pull/823

### CODE REVIEW

File: packages/client-search/src/types/SearchResponse.ts [hunk 1]
```diff
@@ -225,4 +225,6 @@ export type SearchResponse<TObject = {}> = {
    * The relevancy threshold applied to search in a virtual index.
    */
   appliedRelevancyStrictness?: number;
+
+  readonly renderingContent?: Settings['renderingContent'];
```
Comment (eunjae-lee) [inline packages/client-search/src/types/SearchResponse.ts:229]: Why is it `readonly` here? I feel like technically all the fields in this response is readonly, though.

Comment (Haroenv) [reply packages/client-search/src/types/SearchResponse.ts:229]: ah yes it shouldn't be, copied from the other place

Comment (Haroenv) [reply packages/client-search/src/types/SearchResponse.ts:229]:
```suggestion
  renderingContent?: Settings['renderingContent'];
```

**Review by eunjae-lee — APPROVED** [review]

File: packages/client-search/src/types/Rule.ts [hunk 2]
```diff
@@ -125,6 +125,12 @@ export type ConsequenceParams = {
    * Behaves like optionalFilters.
    */
   readonly automaticOptionalFacetFilters?: readonly AutomaticFacetFilter[];
+
+  /**
+   * Content defining how the search interface should be rendered.
+   * A default value for this can be set via settings
+   */
+  readonly renderingContent?: Settings['renderingContent'];
```
Comment (VladislavFitz) [inline packages/client-search/src/types/Rule.ts:133]: I probably missed it, but did we agree on `renderingContent` as a part of `Rule.Consequence`?

Comment (Haroenv) [reply packages/client-search/src/types/Rule.ts:133]: In this object it's part of consequence.params.renderingContent, as we discussed

**Review by VladislavFitz — APPROVED** [review]

**Pull request merged**